azure-iot-device/tests/common/transport/test_pipeline_ops_base.py

```diff
@@ -0,0 +1,112 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+from azure.iot.device.common.transport import pipeline_ops_base
+
+fake_callback = "__fake_callback__"
+fake_feature_name = "__fake_feature_name__"
+
+
+def assert_all_base_defaults(obj):
+    assert obj.name is obj.__class__.__name__
+    assert obj.needs_connection is False
+    assert obj.error is None
+
+
+@pytest.mark.describe("PipelineOperation object")
+class TestPipelineOperation(object):
+    @pytest.mark.it("Sets required and default arguments correctly")
```

Comment:
Okay, so since this seems to be everywhere I won't keep bringing it up after this, but I'd like to see this as a more specific thing that reads like a requirement.